### PR TITLE
Add param force to ipa_user to optionally not set user password

### PR DESCRIPTION
**Pull request by mattymo**
Now ipa_user does not modify user passwords if the account
already exists.

##### SUMMARY

Modifies behavior of ipa_user module when setting passwords. If the account does not exist, the password is set. If the user exists, its password is only set if force=True.

##### ISSUE TYPE

 - Feature Pull Request
##### COMPONENT NAME

ipa_user

##### ANSIBLE VERSION

```
ansible 2.4.2.0
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by mattymo** [comment]
oops. I'll fix syntax issues

**Comment by Akasurde** [comment]
cc @fxfitz

File: lib/ansible/modules/identity/ipa/ipa_user.py [hunk 1]
```diff
@@ -190,6 +195,15 @@ def get_user_diff(client, ipa_user, module_user):
     :param module_user:
     :return:
     """
+    non_updateable_keys = ['force']
```
Comment (fxfitz) [inline lib/ansible/modules/identity/ipa/ipa_user.py]: nit/non-blocking: Have you thought about adding `force` to the docstring?

**Review by fxfitz — APPROVED** [review]

**Comment by fxfitz** [comment]
@mattymo I had no idea ipa_user didn't update the password! Thanks for the contribution!

@Akasurde I'm +1 on this :-)

File: lib/ansible/modules/identity/ipa/ipa_user.py [hunk 2]
```diff
@@ -272,6 +289,7 @@ def main():
     argument_spec = ipa_argument_spec()
     argument_spec.update(displayname=dict(type='str'),
                          givenname=dict(type='str'),
+                         force=dict(type='bool'),
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_user.py]: can you add default value of 'False' to force

```
force=dict(type='bool', default=False),
```

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/identity/ipa/ipa_user.py (hunk 1 above)
Comment (mattymo) [reply lib/ansible/modules/identity/ipa/ipa_user.py]: It is in the docstring on lines 26-30

**Comment by mattymo** [comment]
@fxfitz Actually, it overwrote the password every time and had no option to NOT overwrite (unless you did some ugly things in ansible)

File: lib/ansible/modules/identity/ipa/ipa_user.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/identity/ipa/ipa_user.py]: @mattymo @fxfitz is talking about Function documentation on line 195

**Pull request closed without merging**